### PR TITLE
feat!: Override Parameter for Currencies and Payment Chains in Invoice Dashboard and Create Invoice Form

### PR DESCRIPTION
**Pull request by sstefdev**
Fixes https://github.com/RequestNetwork/web-components/issues/209

### Problem
Currently, the currencies input parameter appends to the default currency list in the Invoice Dashboard and Create Invoice Form. This makes it impossible for app builders to remove a currency from the default list, limiting customization options.

### Changes

BREAKING CHANGE!

- Changed the existing `currencies` parameter to act like an override for the default currency list instead of appending to the currency list on both the Invoice Dashboard and Create Invoice Form components.
- When the `currencies` parameter is passed, only those currencies.
- When the `currencies` parameter is not passed, the default currencies list is used.

### Considerations

App Builders who previously used the `currencies` parameter to _append_ to the default currency list must now treat it like an override by including every currency they wish to support in their currencies argument instead of only the appended currencies.

#### How it looks when we don't have the right currencies in the config, but we have them in the invocies

Screenshot 2024-12-10 at 09 07 31

## Summary by CodeRabbit

- **New Features**
	- Introduced a new SVG icon for representing an unknown state.
	- Enhanced handling of currency display and cipher provider initialization in various components.

- **Bug Fixes**
	- Improved error handling for network and request fetching processes.

- **Chores**
	- Streamlined logic for currency management initialization, enhancing clarity and performance.

### CODE REVIEW

File: shared/utils/initCurrencyManager.ts [hunk 1]
```diff
@@ -67,37 +67,17 @@ import { formattedCurrencyConversionPairs } from './currencyConversionPairs'
...
-  const defaultCurrencies = CurrencyManager.getDefaultList().filter(
-    (currency) => defaultCurrencyIds.includes(currency.id)
-  );
-
-  currenciesToUse = defaultCurrencies;
-
+  // If customCurrencies is provided, use only those
   if (customCurrencies?.length > 0) {
-    currenciesToUse.push(...customCurrencies);
+    return new CurrencyManager(customCurrencies, {}, formattedCurrencyConversionPairs);
```
[the remark this answers is not in the document]
Comment (sstefdev) [reply shared/utils/initCurrencyManager.ts:72]: There is no need for this

**Review by MantisClone — COMMENTED** [review]
@sstefdev Please update the description with usage examples for both the *append* and *override* behavior.

As far as I can tell - the description is incorrect. It mentions a new "append or override" parameter, but this isn't present in the code.

**Comment by MantisClone** [comment]
I added `feat!` to the PR title to indicate this is a BREAKING change.

**Comment by MantisClone** [comment]
I edited the description myself. @sstefdev please review my edits before merging.

**Review by MantisClone — APPROVED** [review]

**Pull request merged**